tests/gdx-tests/src/com/badlogic/gdx/tests/g3d/shadows/system/BaseShadowSystem.java

```diff
@@ -0,0 +1,468 @@
+	public void addLight (PointLight point) {
+		addLight(point, EnumSet.of(CubemapSide.PositiveX, CubemapSide.NegativeX, CubemapSide.PositiveY, CubemapSide.NegativeY,
+			CubemapSide.PositiveZ, CubemapSide.NegativeZ));
+	}
+
+	@Override
+	public void addLight (PointLight point, Set<CubemapSide> sides) {
+		PointLightProperties plProperty = new PointLightProperties();
+		for (int i = 0; i < 6; i++) {
+			CubemapSide cubemapSide = Cubemap.CubemapSide.values()[i];
+			if (sides.contains(cubemapSide)) {
+				PerspectiveCamera camera = new PerspectiveCamera(90, 0, 0);
+				camera.position.set(point.position);
+				camera.direction.set(cubemapSide.direction);
+				camera.up.set(cubemapSide.up);
+				camera.near = 0.1f;
+				camera.far = 1000;
+
+				LightProperties p = new LightProperties(camera);
+				plProperty.properties.put(cubemapSide, p);
+			}
+		}
+		pointCameras.put(point, plProperty);
+	}
+
+	@Override
+	public void removeLight (SpotLight spot) {
+		spotCameras.remove(spot);
+	}
+
+	@Override
+	public void removeLight (DirectionalLight dir) {
+		dirCameras.remove(dir);
+	}
+
+	@Override
+	public void removeLight (PointLight point) {
+		pointCameras.remove(point);
+	}
+
+	@Override
+	public boolean hasLight (SpotLight spot) {
+		if (spotCameras.containsKey(spot)) return true;
+		return false;
+	}
+
+	@Override
+	public boolean hasLight (DirectionalLight dir) {
+		if (dirCameras.containsKey(dir)) return true;
+		return false;
+	}
+
+	@Override
+	public boolean hasLight (PointLight point) {
+		if (pointCameras.containsKey(point)) return true;
+		return false;
+	}
+
+	@Override
+	public void update () {
+		for (ObjectMap.Entry<SpotLight, LightProperties> e : spotCameras) {
+			e.value.camera.position.set(e.key.position);
+			e.value.camera.direction.set(e.key.direction);
+			nearFarAnalyzer.analyze(e.key, e.value.camera, renderableProviders);
+		}
+
+		for (ObjectMap.Entry<DirectionalLight, LightProperties> e : dirCameras) {
+			e.value.camera.direction.set(e.key.direction);
+			directionalAnalyzer.analyze(e.key, e.value.camera, camera).update();
+			e.value.camera.update();
+		}
+
+		for (ObjectMap.Entry<PointLight, PointLightProperties> e : pointCameras) {
+			for (ObjectMap.Entry<CubemapSide, LightProperties> c : e.value.properties) {
+				c.value.camera.position.set(e.key.position);
+				nearFarAnalyzer.analyze(e.key, c.value.camera, renderableProviders);
+			}
+		}
+	}
+
+	@Override
+	public void begin (Camera camera, Iterable<RenderableProvider> renderableProviders) {
+		if (this.renderableProviders != null || this.camera != null) throw new GdxRuntimeException("Call end() first.");
+
+		this.camera = camera;
+		this.renderableProviders = renderableProviders;
+	}
+
+	@Override
+	public void begin (int n) {
+		if (n >= passShaderProviders.length)
+			throw new GdxRuntimeException("Pass " + n + " doesn't exist in " + getClass().getName());
+
+		currentPass = n;
+
+		spotCameraIterator = spotCameras.iterator();
+		spotCameraIterator.reset();
```

Comment:
Calling the `iterator` method always resets it, so you only need one of these two methods.